eng/Packages.Data.props

```diff
@@ -223,7 +223,7 @@
     All should have PrivateAssets="All" set so they don't become package dependencies
   -->
   <ItemGroup>
-    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20240426.2" PrivateAssets="All" />
+    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20240426.1" PrivateAssets="All" />
```

Comment:
Please revert all of your changes under `/eng`.   These should not be here.

Reply:
AutoRest after 3.0.0-beta.20240426.1 breaks our TypeSpec codegen. It's a temporary workaround for us to gen client code correctly. Should I remove it now or I can remove it later after AutoRest issue fixed?

Reply:
I rebase my branch to earlier commit on main instead of revert commit. PR looks clear now.